src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/xml/DocumentUtils.kt

```diff
@@ -29,7 +31,23 @@ object DocumentUtils {
         return null
     }
 
-    private fun getOuCode(item: Node): String? {
+    fun getCourtCode(documents: Element): String? {
+        val sourceFileNameNodes = documents.getElementsByTagName(SOURCE_FILE_NAME)
+        for (j in 0 until sourceFileNameNodes.length) {
+            val sourceFileNameElement = sourceFileNameNodes.item(j) as Element
+            val childTextNodes: NodeList = sourceFileNameElement.childNodes
+            for (k in 0 until childTextNodes.length) {
+                return getOuCode(childTextNodes.item(k))
+            }
+        }
+        return null
+    }
+
+    fun getCourtCode(documents: Element, useXPath: Boolean): String? {
+        return if (useXPath) getCourtCodeByXPath(documents) else getCourtCode(documents)
```

Comment:
Another nice (?) if statement. Not sure about putting it all on one line but I do like the fact that if statements (and other constructs) are expressions and can return values in Kotlin